core/src/androidMain/kotlin/Bluetooth.kt

```diff
@@ -41,7 +46,21 @@ private val Context.locationManager: LocationManager
 private val isLocationEnabled: Boolean
     get() = LocationManagerCompat.isLocationEnabled(applicationContext.locationManager)
 
-internal actual val bluetoothAvailability: Flow<Bluetooth.Availability> =
+private val locationEnabledFlow = when {
+    SDK_INT > R -> flowOf(true)
+    else -> broadcastReceiverFlow(IntentFilter(PROVIDERS_CHANGED_ACTION))
+        .map { intent ->
+            if (SDK_INT == R) {
+                intent.getBooleanExtra(EXTRA_PROVIDER_ENABLED, false)
+            } else {
+                isLocationEnabled
+            }
```

Comment:
Oh, very clever approach (to fallback to `isLocationEnabled` for APIs that don't provide `EXTRA_PROVIDER_ENABLED`). 💯 

So, broadcast receiver for `PROVIDERS_CHANGED_ACTION` still triggers below Android R? Nice. 👍

Reply:
Thank you. I had that in mind before but just did not do it. :-)